authorization/v2/uniformResourceIdentifier.go

```diff
@@ -157,3 +158,51 @@ func marshallURIMap(uriMap map[string]string) string {
 	// Remove the extra '&' character introduced when marshaling the uriMap
 	return marshalledMap[:len(marshalledMap)-1]
 }
+
+// isSubsetOf checks that the URI is a subset of the given URI
+func (uri UniformResourceIdentifier) isSubsetOf(target UniformResourceIdentifier) bool {
+	// Ensure the target path is a subset of the source path
+	if len(uri.path) > len(target.path) {
+		return false
+	}
+
+	// Compare URI path
+	sourcePathComponents := strings.Split(uri.path, ":")
+	targetPathComponents := strings.Split(target.path, ":")
+	for i, pathComponent := range sourcePathComponents {
+		if pathComponent != targetPathComponents[i] {
+			return false
+		}
+	}
+
+	// Validate URI arguments
+	for key, sourceValue := range uri.arguments {
+		if targetValue, ok := target.arguments[key]; ok {
+			match, err := regexp.Match(sourceValue, []byte(targetValue))
+			if !match || err != nil {
+				// Fail-soft, if the regex is invalid or the regex pattern match fails, the URIs don't match
+				return false
+			}
+		} else {
+			// In the case the target URI doesn't contain an argument in the source
+			// the target is no longer a subset of the source.
+			// e.g. Source = hs:hs_auth?test=1 and Target = hs:hs_auth
+			// this means, hs:hs_auth is not a subset of the source since it is limited
```

Comment:
We're not checking if `target` is a subset of `uri` but rather if `uri` is a subset of `target`.

This is not correct. `hs:hs_auth?test=1` is a subset of `hs:hs_auth`, we shouldn't be returning false here.

Reply:
Yes, that's correct. I'll rework how the arguments are compared